sentry-apollo-3/src/main/java/io/sentry/apollo3/SentryApollo3HttpInterceptor.kt

```diff
@@ -114,81 +174,260 @@ class SentryApollo3HttpInterceptor @JvmOverloads constructor(private val hub: IH
             variables?.let {
                 setData("variables", it)
             }
+            setData("http.method", method)
         }
     }
 
-    private fun operationNameFromHeaders(request: HttpRequest): String? {
-        return decodeHeaderValue(request, SENTRY_APOLLO_3_OPERATION_NAME)
-            ?: request.valueForHeader("X-APOLLO-OPERATION-NAME")
-    }
-
     private fun decodeHeaderValue(request: HttpRequest, headerName: String): String? {
-        return request.valueForHeader(headerName)?.let {
+        return getHeader(headerName, request.headers)?.let {
             try {
                 String(Base64.decode(it, Base64.NO_WRAP))
-            } catch (e: IllegalArgumentException) {
-                hub.options.logger.log(SentryLevel.ERROR, "Error decoding internal apolloHeader $headerName", e)
+            } catch (e: Throwable) {
+                hub.options.logger.log(
+                    SentryLevel.ERROR,
+                    "Error decoding internal apolloHeader $headerName",
+                    e
+                )
                 return null
             }
         }
     }
 
-    private fun HttpRequest.valueForHeader(key: String) = headers.firstOrNull { it.name == key }?.value
+    private fun finish(
+        span: ISpan?,
+        request: HttpRequest,
+        response: HttpResponse?,
+        statusCode: Int?,
+        operationName: String?,
+        operationType: String?,
+        operationId: String?
+    ) {
+        var responseContentLength: Long? = null
+        response?.body?.buffer?.size?.ifHasValidLength {
+            responseContentLength = it
+        }
 
-    private fun finish(span: ISpan, request: HttpRequest, response: HttpResponse? = null, statusCode: Int?) {
-        if (beforeSpan != null) {
-            try {
-                val result = beforeSpan.execute(span, request, response)
-                if (result == null) {
-                    // Span is dropped
-                    span.spanContext.sampled = false
+        if (span != null) {
+            statusCode?.let {
+                span.setData("http.response.status_code", statusCode)
+            }
+            responseContentLength?.let {
+                span.setData("http.response_content_length", it)
```

Comment:
nit: a little bit inconsistent that we didn't use a dot here like for `status_code` - `http.response.content_length`, but this is from the spec, so I guess we are good unless we want to alter the spec.